akka-docs/src/test/scala/docs/stream/operators/flow/StatefulMap.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.stream.operators.flow
+import akka.actor.ActorSystem
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+
+import java.util.concurrent.atomic.AtomicInteger
+
+object StatefulMap {
+
+  implicit val actorSystem: ActorSystem = ???
+
+  def indexed(): Unit = {
+    //#indexed
+    Source(List("A", "B", "C", "D"))
+      .statefulMap(() => 0L)((idx, elem) => (idx + 1, (idx, elem)), _ => None)
+      .runWith(Sink.foreach(println))
+    //prints
+    //(0,A)
+    //(1,B)
+    //(2,C)
+    //(3,D)
+    //#indexed
+  }
+
+  def bufferUntilChanged(): Unit = {
+    //#bufferUntilChanged
+    Source("A" :: "B" :: "B" :: "C" :: "C" :: "C" :: "D" :: Nil)
+      .statefulMap(() => List.empty[String])(
+        (buffer, elem) =>
+          buffer match {
+            case Nil                       => (elem :: Nil, Nil)
+            case head :: _ if head != elem => (elem :: Nil, buffer.reverse)
```

Comment:
Reverse isn't really important for a list of identical strings, so maybe just emit the buffer here and in the end?